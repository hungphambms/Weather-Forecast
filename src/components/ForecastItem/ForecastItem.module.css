.item {
  background: rgba(255, 255, 255, .87);

  border: var(--color-light) 1px solid;
  border-radius: calc(var(--border-radius) * 3);
  padding: 16px 16px 8px;
  margin-bottom: 12px;

  position: relative;
}

.item h3 {
  font-size: 18px;
  padding: 0;
  margin: 0 0 8px;
  font-weight: 500;
  color: var(--color-dark);
}

.item p {
  font-size: 14px;
  padding: 0;
  margin: 0 0 8px;
  color: var(--color-medium);
}

.item label {
  position: absolute;
  top: calc(50% - 24px);
  right: 24px;

  font-size: 32px;
  color: var(--color-success);
}

.item p strong {
    color: var(--color-dark);
}


.item p img {
  width: 24px;
  margin-right: 8px;
  position: relative;
  top: -4px;
}

.divider {
  display: inline-block;
  margin: 0 16px;
}
