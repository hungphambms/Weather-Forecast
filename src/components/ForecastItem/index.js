// import React from "react";
import PropTypes from "prop-types";

import Moment from "react-moment";
import { k2c } from "../../utils/temperature";

import styles from "./ForecastItem.module.css";

import imgMin from "./images/min.png";
import imgMax from "./images/max.png";

function ForecastItem({ item }) {
  return (
    <div aria-label="item" className={styles.item}>
      <h3>
        <Moment date={item.dt * 1000} format="dddd" />
      </h3>

      <p>
        <Moment date={item.dt * 1000} format="MMM DD, YYYY" />
      </p>

      <p>
        <img src={imgMin} alt="min icon" />
        <strong>Min</strong> {k2c(item.temp.min)}
        <span className={styles.divider}></span>
        <img src={imgMax} alt="max icon" />
        <strong>Max</strong> {k2c(item.temp.max)}
      </p>

      <label>
        <img
          src={`http://openweathermap.org/img/w/${item.weather[0].icon}.png`}
          alt="Temp Icon"
        />

        {k2c(item.temp.day)}
      </label>
    </div>
  );
}

ForecastItem.defaultProps = {
  item: {},
};

ForecastItem.propTypes = {
  item: PropTypes.object,
};

export default ForecastItem;
