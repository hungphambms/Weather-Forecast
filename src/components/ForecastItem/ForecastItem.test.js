import { render, cleanup, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';

import ForecastItem from './index';

const SAMPLE_PROPS = {
  dt: 1615006800,
  sunrise: 1614985544,
  sunset: 1615028631,
  temp: {
    day: 309.15,
    min: 295.47,
    max: 309.15,
    night: 297.99,
    eve: 301.63,
    morn: 295.47,
  },
  feels_like: {
    day: 309.14,
    night: 299.5,
    eve: 300.95,
    morn: 297.64,
  },
  pressure: 1013,
  humidity: 28,
  dew_point: 288.26,
  wind_speed: 2.11,
  wind_deg: 64,
  weather: [
    {
      id: 800,
      main: "Clear",
      description: "clear sky",
      icon: "01d",
    },
  ],
  clouds: 0,
  pop: 0,
  uvi: 13.87,
};

afterEach(cleanup);

describe('ForecastItem rendering', () => {
  it('renders ForecastItem without crashing', () => {
    const render = renderer.create(<ForecastItem item={SAMPLE_PROPS} />);
    const tree = render.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correct items', async() => {
    const component = render(<ForecastItem item={SAMPLE_PROPS} />);
    expect(component).not.toBeNull();
  });
});
