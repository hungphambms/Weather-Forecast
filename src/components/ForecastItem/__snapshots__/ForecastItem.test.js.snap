// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem rendering renders ForecastItem without crashing 1`] = `
<div
  aria-label="item"
  className="item"
>
  <h3>
    <time
      dateTime={"2021-03-06T05:00:00.000Z"}
    >
      Saturday
    </time>
  </h3>
  <p>
    <time
      dateTime={"2021-03-06T05:00:00.000Z"}
    >
      Mar 06, 2021
    </time>
  </p>
  <p>
    <img
      alt="min icon"
      src="min.png"
    />
    <strong>
      Min
    </strong>
     
    22
    <span
      className="divider"
    />
    <img
      alt="max icon"
      src="max.png"
    />
    <strong>
      Max
    </strong>
     
    36
  </p>
  <label>
    <img
      alt="Temp Icon"
      src="http://openweathermap.org/img/w/01d.png"
    />
    36
  </label>
</div>
`;
